### PR TITLE
Added Xcode 12.5.1 to tests and also made this version the default one.

### PR DESCRIPTION
**Pull request by apps4everyone**
Added Xcode 12.5.1 to tests and also made this version the default one.

### CODE REVIEW

**Review by danyf90 — CHANGES_REQUESTED** [review]
The macos latest GitHub executor does not contain Xcode 12.5.1.
On the other and, the macos.11 executor contains 12.5.1 but doesn't contain 12.2

**Comment by apps4everyone** [comment]
> The [macos latest (alias for macos 10.15)](https://github.com/actions/virtual-environments/blob/main/images/macos/macos-10.15-Readme.md) GitHub executor does not contain Xcode 12.5.1.
> On the other and, the [macos 11](https://github.com/actions/virtual-environments/blob/main/images/macos/macos-11-Readme.md#xcode) executor contains 12.5.1 but doesn't contain 12.2

thx, for the info. Removed 12.2 and added 13. Only to see if tests would be ok.

**Comment by danyf90** [comment]
@apps4everyone you still need to change the `runs-on: macOS-latest` to `runs-on: macOS-11` to use the newer image. 
I know, it's called latest but it's not really the latest :D

File: .github/workflows/tuist.yml [hunk 1]
```diff
@@ -60,10 +60,10 @@ jobs:
           ./fourier test tuist unit
   release_build:
     name: Release build with Xcode ${{ matrix.xcode }}
-    runs-on: macOS-latest
+    runs-on: macOS-11
     strategy:
       matrix:
-        xcode: ['12.2', '12.4'] # keep 12.2 to ensure backwards compatibility
+        xcode: ['12.4', '12.5.1', '13.0'] # keep 12.2 to ensure backwards compatibility
```
Comment (danyf90) [inline .github/workflows/tuist.yml:66]:
```suggestion
        xcode: ['12.4', '12.5.1', '13.0'] # keep 12.4 to ensure backwards compatibility
```

Comment (danyf90) [inline .github/workflows/tuist.yml:66]: @fortmarek @pepibumur @kwridan Do we need to keep 12.2 compatibility? if yes, we should setup another workflow to run on the 10.15 executor

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,11 @@ Please, check out guidelines: https://keepachangelog.com/en/1.0.0/
 
 - Allow to pass Cloud authentication token via TUIST_CLOUD_TOKEN even when not CI [#3380](https://github.com/tuist/tuist/pull/3380) by [@danyf90](https://github.com/danyf90)
 
+
+### Changed
+
+- **Breaking** Minimum supported Xcode version bumped to 12.4.
```
Comment (danyf90) [inline CHANGELOG.md:17]: Not related to this PR, but could you please merge the 2 `Fixed` section? 🙏

Comment (apps4everyone) [reply CHANGELOG.md:9]: done

**Review by danyf90 — APPROVED** [review]

File: .github/workflows/tuist.yml (hunk 1 above)
Comment (fortmarek) [reply .github/workflows/tuist.yml]: Considering 13.0 is right around the corner, I think it should be fine to sunset 12.2

**Comment by fortmarek** [comment]
I'd like to fix the CI issue, so closing this and superseded by https://github.com/tuist/tuist/pull/3427.

@apps4everyone I added you as a contributor, so you don't have to create a separate fork next time you do a PR.

**Pull request closed without merging**